src/code/sys_cmpdma.c

```diff
@@ -1,32 +1,117 @@
 #include "global.h"
 #include "system_malloc.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_cmpdma/func_80178750.s")
+typedef struct {
+    union {
+        u32 dmaWord[2];
+        u32 dataStart;
+        u32 dataSize;
+        struct {
+            u32 start;
+            u32 end;
+        } offset;
+    };
+} CmpDmaBuffer;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_cmpdma/func_80178818.s")
+static CmpDmaBuffer sDmaBuffer;
```

Comment:
Could you remove the `static` on this bss variable so it appears on the map file?
```suggestion
CmpDmaBuffer sDmaBuffer;
```